brave/browser-laptop
# Create sectionTitle.js

TODO: Refactor modalOverlay.js to remove `.sectionTitle`. ~~This requires #8038~~ merged

**This PR blocks https://github.com/brave/browser-laptop/pull/8044.**

Closes #7750
Closes #8165
Closes #8234
Addresses #8235
Closes #8349
Closes #8350

Created `TitleWrapper`, `SectionTitle`, and `SubTitle`
 - SectionTitleWrapper and BetaSectionTitleWrapper
 - DefaultSectionTitle, PrefSectionTitle, and AboutPagesSectionTitle
 - AboutPagesSectionSubTitle and SectionSubTitle

Refactored with those components:
 - extensionsTab.js, pluginsTab.js, preferences.js, syncTab.js, and paymentsTab.js
 - about.js, brave.js, bookmarks.js and history.js
 - torrentViewer.js, torrentStatus.js and torrentFileList.js

Added `siteDetailsPageContent` to commonStyles.js and applied:
 - about.js, brave.js, and history.js

Removed styles of sectionTitle from:
 - brave.less
 - siteDetails.less

Auditors:

Test Plan 1:
1. Open about:preferences#search
2. Make sure there is margin under the top header

Test Plan 2:
1. Open about:preferences#sync
2. Make sure "beta" is aligned top
3. Open about:preferences#payments
4. Make sure the labels appear the same row
5. Make sure the switches and the labels are aligned exactly like #7751

Test Plan 3:
1. Open about:about
2. Make sure there is margin under the header
3. Open about:brave, about:bookmarks, and about:history
4. Make sure the header font size is same

Test Plan 4:
1. Open https://webtorrent.io/free-torrents/
2. Click .torrent link
3. Make sure the header font size is same as about:brave
4. Start download
5. Make sure the header font size is same

Test Plan 5:
1. Clear your profile folder
2. Open about:preferences#sync
3. Enable Sync
4. Make sure there is margin under "Devices"

Test Plan 6:
1. Open "Import Browser data" from the menu
2. Make sure the switch is indented

Test Plan 7:
1. Open about:styles
2. Click "commonForm" on TOC
3. Make sure the template of the commonForm is displayed
4. Click "Go back to top"
5. Click "sectionTitle" on TOC
6. Make sure section title components are displayed


Test Plan:
